basic01-xdp-pass/README.org

```diff
@@ -121,21 +121,30 @@ the standard =ip= tool; so in this case you can actually load our ELF-file
 =xdp_pass_kern.o= (where we named our ELF section "xdp") like this:
 
 #+begin_example sh
- ip link set dev lo xdpgeneric obj xdp_pass_kern.o sec xdp
+$ sudo ip link set dev lo xdpgeneric obj xdp_pass_kern.o sec xdp
 #+end_example
 
 Listing the device via =ip link show= also shows the XDP info:
 
 #+begin_example sh
-$ ip link show dev lo
+$ sudo ip link show dev lo
 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 xdpgeneric qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
     link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
     prog/xdp id name xdp_prog_simple 220 tag 3b185187f1855c4c jited
 #+end_example
 
+Should you run it without =sudo=, you would have less information:
+
+#+begin_example sh
+$ ip link show dev lo
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 xdpgeneric qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    prog/xdp id name
```

Comment:
Something is off here, it will not show `id name` it will show the numeric id (`id 220` if using the example above). Looks like the output has been similarly mangled in the example run as root above, so could you please fix it up there as well (should be ` prog/xdp id 220 name xdp_prog_simple tag 3b185187f1855c4c jited`).

Reply:
side comment: the name requires an updated iproute.
The former trace is using Ubuntu's 22.04:
```
ip -V
ip utility, iproute2-5.15.0, libbpf 0.5.0
```

then using a recent iproute2:
```
$ sudo ./ip link show dev lo
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 xdpgeneric qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    prog/xdp id 408 name xdp_prog_simple tag 3b185187f1855c4c jited 
$ ./ip -V
ip utility, iproute2-6.7.0, libbpf 0.5.0
```

I'll assume that iproute2 has been updated since moving forward it'll be the case for anyone who will follow this README.

Reply:
done by the commit [1b8d68a](https://github.com/xdp-project/xdp-tutorial/pull/397/commits/1b8d68a3c3ed805fbf72f6fd2763e1016ff4c812)